packages/headless/src/controllers/facets/range-facet/date-facet/headless-date-filter.ts

```diff
@@ -0,0 +1,203 @@
+import {SearchEngine} from '../../../../app/search-engine/search-engine';
+import {
+  ConfigurationSection,
+  DateFacetSection,
+  SearchSection,
+} from '../../../../state/state-sections';
+import {loadReducerError} from '../../../../utils/errors';
+import {
+  buildController,
+  Controller,
+} from '../../../controller/headless-controller';
+import {configuration, dateFacetSet, search} from '../../../../app/reducers';
+import {determineFacetId} from '../../_common/facet-id-determinor';
+import {DateFacetValue} from '../../../../features/facets/range-facets/date-facet-set/interfaces/response';
+import {deselectAllFacetValues} from '../../../../features/facets/facet-set/facet-set-actions';
+import {updateFacetOptions} from '../../../../features/facet-options/facet-options-actions';
+import {executeSearch} from '../../../../features/search/search-actions';
+import {
+  logFacetClearAll,
+  logFacetSelect,
+} from '../../../../features/facets/facet-set/facet-set-analytics-actions';
+import {
+  registerDateFacet,
+  RegisterDateFacetActionCreatorPayload,
+  updateDateFacetValues,
+} from '../../../../features/facets/range-facets/date-facet-set/date-facet-actions';
+import {validateDateFacetOptions} from './headless-date-facet-options';
+import {dateFacetResponseSelector} from '../../../../features/facets/range-facets/date-facet-set/date-facet-selectors';
+
+/**
+ * The options defining a `DateFilter`.
+ */
+export interface DateFilterOptions {
+  /**
+   * The field whose values you want to display in the filter.
+   */
+  field: string;
+
+  /**
+   * A unique identifier for the controller.
+   * By default, a unique random identifier is generated.
+   */
+  facetId?: string;
```

Comment:
From a user perspective, I still find it very confusing that `facet` is used here, given that this component (seemingly) has nothing to do with facets.

Reply:
Despite the name, it's meant to be used as a facet, with other facets.
Imagine the following component 
![Screen Shot 2021-06-22 at 2 56 44 PM](https://user-images.githubusercontent.com/4923043/122983505-11967800-d36a-11eb-89b3-a12e1f8db071.png)

Or perhaps a slider

Reply:
It also has the ability to return the numberOfValues, which is why filterFacetCount / injectionDepth are present

Reply:
I understand that this is the way we use it in Atomic, but this component would also be what's used if a client wants to manually specify a range from a different UI component.

Reply:
Yes will all facets, which are technically field values with counts associated, they can be used in any UI components as well

Reply:
Isn't the goal of a facet to filter results? I'm not sure why we should consider this controller not to be a facet?